pil_analyzer/src/pil_analyzer.rs

```diff
@@ -83,7 +84,9 @@ impl<T> From<PILAnalyzer<T>> for Analyzed<T> {
         };
         result.pre_visit_expressions_mut(&mut |e| {
             if let Expression::Reference(Reference::Poly(reference)) = e {
-                assign_id(reference);
+                if !constant_names.contains(&reference.name) {
```

Comment:
Think I asked this on #647 too, but shouldn't `Reference` have a `Constant` variant? For example, in other places you also check `if !poly.next && poly.index.is_none()`, shouldn't that be checked once when the reference is created and then stored with the reference?

Reply:
I would like to remove the distinction between `constant` and other symbols. that's why `reference` does not have it. The `next` and `index` will also be replaced by proper operators in the future, so these checks will disappear.

Reply:
I have not decided yet what to do here. This line here is actually more like a bugfix of existing code.

Reply:
I haven't thought it through, but the struct will certainly be cleaned up at the end of my refactoring.